### PR TITLE
NameService: minor improvements, part 1

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Refactor code a bit and make it closer to https://datatracker.ietf.org/doc/html/rfc1034.

### CODE REVIEW

**Comment by roman-khimov** [comment]
Just to give some background, we have an NNS contract in (https://github.com/nspcc-dev/neofs-contract/), it was originally based on C# one, but it diverged over time as we solved some problems related to NeoFS CDN and dApp needs. This contract is used in NeoFS and it also powers dApp examples shown on Aug 30, so it's tested and known to work. We think changes made there are very much relevant for real-life applications and standard public NNS contract will benefit from them.

**Comment by roman-khimov** [comment]
@shargon, @superboyiii, @erikzhang? This part is rather trivial, but we have more invasive things in the queue, so maybe we can deal with this one first and then discuss other things.

File: src/NameService/NameService.cs [hunk 1]
```diff
@@ -353,6 +355,7 @@ public static string GetRecord(string name, RecordType type)
             token.EnsureNotExpired();
```
Comment (erikzhang) [inline src/NameService/NameService.cs:355]: Returns `null` if `nameMap[tokenKey]` is `null`?

File: src/NameService/NameService.cs [hunk 2]
```diff
@@ -463,23 +470,38 @@ private static void PostTransfer(UInt160 from, UInt160 to, ByteString tokenId, o
...
-            for (int i = 1; i < root.Length; i++)
+            for (int i = 1; i < root.Length - 1; i++)
             {
                 c = root[i];
-                if (!(c >= 'a' && c <= 'z' || c >= '0' && c <= '9')) return false;
+                if (!(isAlphaNum(c) || c == '-')) return false;
             }
-            return true;
+            c = root[root.Length - 1];
+            return isAlphaNum(c);
```
Comment (erikzhang) [inline src/NameService/NameService.cs:492]: The case of `root.Length == 1` needs to be considered.

File: src/NameService/NameService.cs (hunk 1 above)
Comment (AnnaShaleva) [reply src/NameService/NameService.cs:355]: It is fixed in #25, see the https://github.com/neo-project/non-native-contracts/pull/25/commits/20d34ea52106c6d2a922b435a90c5f1be6355940.

File: src/NameService/NameService.cs (hunk 2 above)
Comment (AnnaShaleva) [reply src/NameService/NameService.cs:492]: Technically, this case was handled by the existing code, but I've added the explicit `root.Length == 1` check. Fixed.

**Review by roman-khimov — APPROVED** [review]
Everything solved, looks beautiful, good to go. Let's merge this one and move on to the part 2.

File: src/NameService/NameService.cs (hunk 1 above)
Comment (shargon) [reply src/NameService/NameService.cs:355]: Move this fix to this PR?

Comment (AnnaShaleva) [reply src/NameService/NameService.cs:355]: There's a lot of other code refactoring made in https://github.com/neo-project/non-native-contracts/commit/20d34ea52106c6d2a922b435a90c5f1be6355940 which is related GetRecord/GetRecords/GetAllRecords, can we keep it inside a separate PR? We can discuss these changes separately in #25.

Comment (roman-khimov) [reply src/NameService/NameService.cs:355]: I'd really prefer going step by step. The changes in this PR are simple and internally consistent to me, the problem raised is not related to these changes, it always existed and we'll deal with it in #25.

**Review by erikzhang — APPROVED** [review]

**Pull request merged**